src/circuit/uint32.rs

```diff
@@ -671,6 +671,19 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_uint32_shr_overflow() {
+        let mut rng = XorShiftRng::from_seed([0x5dbe6259, 0x8d313d76, 0x3237db17, 0xe5bc0654]);
+
+        for _ in 0..50 {
```

Comment:
Is it really needed to do this 50 times? The behaviour is easily seen from the source to not be value-dependent.